src/NuGet.Clients/PackageManagement.PowerShellCmdlets/NuGetPowerShellBaseCommand.cs

```diff
@@ -276,80 +276,62 @@ protected override sealed void ProcessRecord()
+            if (string.IsNullOrEmpty(source))
             {
-                // Convert a relative local URI into an absolute URI
-                source = ConvertRelativeUriToAbsolute(source);
-
-                // Convert source into a PackageSource
-                var packageSource = new PackageSource(source);
+                source = (string)GetPropertyValueFromHost(ActivePackageSourceKey);
+            }
+            
+            // Look through all available sources (including those disabled) by matching source name and URL (or path).
+            var matchingSource = GetMatchingSource(source);
+            if (matchingSource != null)
+            {
+                return SourceValidationResult.Valid(
+                    source,
+                    _sourceRepositoryProvider?.CreateRepository(matchingSource));
+            }
 
-                // Check if the source is a valid http
-                if ((packageSource.IsHttp && packageSource.TrySourceAsUri == null))
-                {
-                    throw new PackageSourceException(PackageSourceException.ExceptionType.UnknownSource);
-                }
+            // If we really can't find a source string, return an empty validation result.
+            if (string.IsNullOrEmpty(source))
+            {
+                return SourceValidationResult.None;
             }
-            return source;
+
+            return CheckSourceValidity(source, validateSource);
         }
 
+        protected void UpdateActiveSourceRepository(string source, bool validateSource)
+        {
+            var result = ValidateSource(source, validateSource);
+
+            UpdateActiveSourceRepository(result);
+        }
 
         /// <summary>
         /// Initializes source repositories for PowerShell cmdlets, based on config, source string, and/or host active source property value.
         /// </summary>
         /// <param name="source">The source string specified by -Source switch.</param>
-        /// <param name="matchingSource">Matching Source if the specified source is a known source. Else null.</param>
-        protected void UpdateActiveSourceRepository(string source, bool validateSource)
+        protected void UpdateActiveSourceRepository(SourceValidationResult result)
         {
-            // If source string is not specified, get the current active package source from the host
-            source = string.IsNullOrEmpty(source) ? (string)GetPropertyValueFromHost(ActivePackageSourceKey) : source;
-
-            // Look through all available sources (including those disabled) by matching source name and url
-            var matchingSource = GetMatchingSource(source);
-            
-            // Check if the source is valid http, local. Else throw an exception.
-            if (validateSource && matchingSource == null)
+            if (result.Validity == SourceValidity.UnknownSource)
```

Comment:
It's not clear the purpose of intermediate "result" structure if it throws here anyway. The same way `ValidateSource` could throw when source is unknown.